tests/e2e/utils/prometheus/setup_prometheus_server.py

```diff
@@ -0,0 +1,152 @@
+
+def set_up_prometheus_port_forwarding():
+    def callback():
+        get_pod_name_command = 'kubectl get pods --namespace=monitoring'.split()
+        pod_list = subprocess.check_output(get_pod_name_command, encoding="utf-8").split('\n')[1:]
+        prometheus_pod_name = None
+        for pod in pod_list:
+            pod_name = pod.split()[0]
+            print("pod name and status:", pod)
+            pod_name_array = pod_name.split('-')
+            if (('p' == pod_name[0])
+                and ('prometheus' == pod_name_array[0])
+                and ('deployment' == pod_name_array[1])):
+                print("Made it into inner if loop.")
+                describe_pod_command = f'kubectl describe pod {pod_name} --namespace monitoring'.split()
+                print('\n\nAbout to call describe pod command!\n\n')
+                print(subprocess.check_output(describe_pod_command, encoding="utf-8"))
+                print('\n\nDescribe Finished\n\n')
+                get_ns_pod_logs_command = f'kubectl logs {pod_name} --namespace monitoring'.split()
+                print("\n\nAbout to get monitoring namespace logs!\n\n")
+                print(subprocess.check_output(get_ns_pod_logs_command, encoding="utf-8"))
+                print('\n\nLogs Finished\n\n')
+                pod_status = pod.split()[1]
+                print("pod_status:", pod_status)
+                assert "1/1" == pod_status
+                print("Assert running was succesfull.")
+                prometheus_pod_name = pod_name
+                break
+        return prometheus_pod_name
+    
+    prometheus_pod_name = wait_for(callback)
+    if None == prometheus_pod_name:
+        raise ValueError("Prometheus Pod Not Running.")
+    print("About to assemble port-forwarding command.")
+    set_up_port_forwarding_command = f'kubectl port-forward {prometheus_pod_name} 9090:9090 -n monitoring'.split()
+    print("prometheus_pod_name:", prometheus_pod_name)
+    print(" ".join(set_up_port_forwarding_command))
+    subprocess.Popen(set_up_port_forwarding_command)
+    time.sleep(10)  # Wait 10 seconds for port forwarding to open
+    
+def check_prometheus_is_running():
+    check_up_command = 'curl http://localhost:9090/api/v1/query?query=up'.split()
+    print(' '.join(check_up_command))
+    up_results = subprocess.check_output(check_up_command, encoding="utf-8")
+    assert True==bool(up_results)
+
+def check_AMP_connects_to_prometheus(region, workspace_id, expected_value):
+    time.sleep(30) # Wait for promehtue sto scrape.
+    access_key = os.environ['AWS_ACCESS_KEY_ID']
+    print("ACCESS_KEY:", access_key)
+    secret_key = os.environ['AWS_SECRET_ACCESS_KEY']
+    print("SECRET_KEY:", secret_key)
+    
+    AMP_awscurl_command = f'awscurl --access_key {access_key} --secret_key {secret_key} --region {region} --service aps https://aps-workspaces.{region}.amazonaws.com/workspaces/{workspace_id}/api/v1/query?query=experiment_server_create_requests'.split()
```

Comment:
2 things:

1. Will this work with temporary credentials?
2. Lets not print the access_key and secret_key for security reasons